scripts/linters/pylint_extensions.py

```diff
@@ -1858,6 +1858,37 @@ def process_tokens(self, tokens):
                         line=line_num)
 
 
+class BlankLineBelowFunctionDefChecker(checkers.BaseChecker):
+    """Custom pylint checker which ensures that there isn't a blank space
+    below a function definition.
+    """
+
+    __implements__ = interfaces.IAstroidChecker
+    name = 'blank-line-below-function-definition'
+    priority = -1
+    msgs = {
+        'C0030': (
+            'Blank line not permitted below function definition',
+            'blank-line-below-function-definition',
+            'The Blank line below the function definition should be removed',
+        ),
+    }
+
+    def visit_functiondef(self, node):
+        """Visit every function definition in a module and check if there is a
+        blank line below the function definition.
+
+        Args:
+            node: astroid.nodes.FunctionDef. Node for a function definition
+                in the AST.
+        """
+        line_number = node.fromlineno
+        line_number += 1
+        if linecache.getline(node.root().file, line_number).strip() == b'':
+            self.add_message(
+                'blank-line-below-function-definition', node=node)
+
+
```

Comment:
Can we use this check to find whether there's a newline in the first line of function definition?

1. functionDefNode.lineno + len(functionDefNode.doc.split('\n')) != functionDefNode.child.lineno

Reply:
Thanks for the suggestion. Ill check and get back.